pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -69,14 +80,35 @@ func (ca *cloudCIDRAllocator) performMultiNetworkCIDRAllocation(node *v1.Node, i
 			if err != nil {
 				return nil, err
 			}
-			if resourceName(inf.Network) != resourceName(gnp.Spec.VPC) || resourceName(inf.Subnetwork) != resourceName(gnp.Spec.VPCSubnet) {
+
+			// Network object must match GNP on network attachment OR on (VPC + subnet).
+			// Otherwise, proceed to next network.
+			matchVPC := resourceName(inf.Network) == resourceName(gnp.Spec.VPC)
+			matchSubnet := resourceName(inf.Subnetwork) == resourceName(gnp.Spec.VPCSubnet)
+			// Don't match empty network attachment names.
+			matchNetworkAttachment := inf.NetworkAttachment != "" && resourceName(inf.NetworkAttachment) == resourceName(gnp.Spec.NetworkAttachment)
+			match := matchNetworkAttachment || (matchVPC && matchSubnet)
+			if !match {
 				continue
 			}
 			klog.V(2).InfoS("interface matched, proceeding to find a secondary range", "nodeName", node.Name, "networkInterface", inf.Name)
+
 			// TODO: Handle IPv6 in future.
-			var secondaryRangeNames []string
-			if gnp.Spec.PodIPv4Ranges != nil {
-				secondaryRangeNames = gnp.Spec.PodIPv4Ranges.RangeNames
+
+			// When Network Attachment is specified, get the interface's alias IP range
+			// for pods from rangeNameAliasIPMap.
+			netAttachment := gnp.Spec.NetworkAttachment
+			if netAttachment != "" {
```

Comment:
I feel the same `if` block is repeated multiple times, one for network attachment and one for secondary ranges. In both places, we are effectively looking up a key in a map and updating annotations etc. 

You can rename variables and avoid duplication. For example, `secondaryRangeNames` below --> `podRanges` and then populating that array depending on whether podRanges are set OR network attachment is set.

Feel free to come up with alternative better names.

Reply:
Done.